java/src/main/java/com/saucelabs/saucebindings/SauceMobileOptions.java

```diff
@@ -0,0 +1,175 @@
+            "commandTimeouts",
+            "crosswalkApplication",
+            "autoGrantPermissions",
+            "enableAnimations",
+            "name");
+
+    public SauceMobileOptions() {
+        this(new MutableCapabilities());
+    }
+
+    // TODO: require users to work with Appium's MobileOptions class similar to Selenium
+    public SauceMobileOptions(MutableCapabilities options) {
+        appiumCapabilities = new MutableCapabilities(options.asMap());
+    }
+
+    public MutableCapabilities toCapabilities(DataCenter dataCenter) {
+        if (app != null) {
+            browserName = null;
+        }
+
+        mobileW3COptions.forEach((capability) -> {
+            addCapabilityIfDefined(appiumCapabilities, capability);
+        });
+
+        if (dataCenter.supportsW3C()) {
+            useW3cCapabilities();
+        } else {
+            useJwpCapabilities(dataCenter.isTestObject());
+        }
+        return appiumCapabilities;
+    }
+
+    private void useW3cCapabilities() {
+        MutableCapabilities sauceCapabilities = new MutableCapabilities();
+        useSaucePlatform(sauceCapabilities);
+
+        appiumDefinedOptions.forEach((capability) -> {
+            addAppiumCapabilityIfDefined(appiumCapabilities, capability);
+        });
+
+        mobileSauceDefinedOptions.forEach((capability) -> {
+            addCapabilityIfDefined(sauceCapabilities, capability);
+        });
+
+        sauceDefinedOptions.forEach((capability) -> {
+            addCapabilityIfDefined(appiumCapabilities, capability);
+        });
+
+        appiumCapabilities.setCapability("sauce:options", sauceCapabilities);
+    }
+
+    private void useJwpCapabilities(boolean to) {
+        if (to) {
+            appiumCapabilities.setCapability("testobject_api_key", getTestObjectKey());
+            if (deviceName == null) {
+                this.deviceName = "Google Pixel XL";
+            }
+        } else {
+            useSaucePlatform(appiumCapabilities);
+        }
+
+        appiumDefinedOptions.forEach((capability) -> {
+            addCapabilityIfDefined(appiumCapabilities, capability);
+        });
+
+        mobileSauceDefinedOptions.forEach((capability) -> {
+            addCapabilityIfDefined(appiumCapabilities, capability);
+        });
+
+        realDeviceSauceDefinedOptions.forEach((capability) -> {
+            addCapabilityIfDefined(appiumCapabilities, capability);
+        });
+    }
+
+    private void useSaucePlatform(MutableCapabilities capabilities) {
+        if (deviceName == null) {
+            this.deviceName = "Android GoogleAPI Emulator";
+        }
+        addAuthentication(capabilities);
+    }
+
+    private void addAppiumCapabilityIfDefined(MutableCapabilities capabilities, String capability) {
+        Object value = getCapability(capability);
+        if (value != null) {
+            capabilities.setCapability("appium:" + capability, value);
+        }
+    }
+
+    String getTestObjectKey() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=5346038)